File: frameworks/Java/restexpress/src/main/java/hello/controller/JsonController.java
package hello.controller;

import org.restexpress.Request;
import org.restexpress.Response;

public class JsonController {

	public HelloWorld sayHello(Request request, Response response) {
		return new HelloWorld();
	}

	/**
	 * Inner class just to illustrate JSON serialization.
	 */
	public static class HelloWorld {
		@SuppressWarnings("unused")
		private String message = "Hello, World!";
	}
}

File: frameworks/PHP/symfony/src/Controller/JsonController.php
<?php

namespace App\Controller;

use Symfony\Component\HttpFoundation\JsonResponse;
use Symfony\Component\Routing\Annotation\Route;

class JsonController
{
    /**
     * @Route("/json")
     */
    public function json(): JsonResponse
    {
        return new JsonResponse(['message' => 'Hello, World!']);
    }
}
